builder/parser/line_parsers.go

```diff
@@ -107,12 +107,15 @@ func parseNameVal(rest string, key string) (*Node, map[string]bool, error) {
 				quote = ch
 				blankOK = true
 				phase = inQuote
-				continue
```

Comment:
@tiborvass if you look at https://github.com/duglin/docker/blob/Issue10391/builder/parser/line_parsers.go#L109 you'll see that when we see a quote we change our "phase" and we save the quote character (in the 'quote' variable), but then we fall thru to https://github.com/duglin/docker/blob/Issue10391/builder/parser/line_parsers.go#L109 which adds the current character ( which is the quote) to the word.
So, short answer, shellParser will remove the quotes. We needed to move it from here because later on when we do the ${} processing we have to know whether there are single vs double quotes around things to know whether to do the substitution. So this func is now just responsible for breaking things into words.